app/src/main/java/com/example/habitsmasher/AddPictureDialog.java

```diff
@@ -0,0 +1,204 @@
+    private Button _cancelButton;
+    private PictureSelectionUser _parentDialog;
+    private ImageButton _galleryButton;
+    private ImageButton _cameraButton;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        // Inflate view and attach elements
+        View view = inflater.inflate(R.layout.add_picture_dialog, container,false);
+        initializeUIElements(view);
+
+        // Set header
+        _header.setText("Select Picture");
+
+        // Set tag
+        TAG = "AddPictureDialogue";
+
+        // Set Cancel button
+        setCancelButtonListener();
+
+        // Set gallery and camera buttons
+        setGalleryButtonListener();
+        setCameraButtonListener();
+
+        return view;
+    }
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        try {
+            // Get parent fragment to send image back to
+            _parentDialog = (PictureSelectionUser) getTargetFragment();
+        }
+        catch (ClassCastException e) {
+            Log.e(TAG, "Exception" + e.getMessage());
+        }
+    }
+
+    /**
+     * Initialize the UI elements of this dialog
+     * @param view
+     */
+    private void initializeUIElements(View view) {
+        _header = view.findViewById(R.id.add_picture_header_text);
+        _cancelButton = view.findViewById(R.id.select_image_cancel_button);
+        _galleryButton = view.findViewById(R.id.gallery_button);
+        _cameraButton = view.findViewById(R.id.camera_button);
+    }
+
+    /**
+     * Reference: https://stackoverflow.com/questions/10165302/dialog-to-pick-image-from-gallery-or-from-camera
+     * Override onActivityResult to handle when user has selected image
+     * @param requestCode
+     * @param resultCode
+     * @param imageReturnedIntent
+     */
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent imageReturnedIntent) {
+        super.onActivityResult(requestCode, resultCode, imageReturnedIntent);
+        switch (requestCode){
+            case 1:
+                // If the user selects the image from the gallery
+                if (resultCode == RESULT_OK) {
+                    // Set selected picture
+                    _selectedImage = imageReturnedIntent.getData();
+                    _parentDialog.setImage(_selectedImage);
+
+                    // Close dialog
+                    this.dismiss();
+                }
+                break;
+            case 2:
+                // If the user takes the image it will already be stored in _selectedImage
+                if (resultCode == RESULT_OK) {
+                    // Set selected picture
+                    _parentDialog.setImage(_selectedImage);
+
+                    // Close dialog
+                    this.dismiss();
+                }
+                break;
+        }
```

Comment:
Just a question, if we are doing conditional checks on the `resultCode`, why do we need the `switch-case` statement? Can we not just go right to the `if` statements?

Reply:
The switch-case statement is for the requestCode (slightly different). The first case is essentially the gallery case, while the second one is the camera case. Both still require the result code to be RESULT_OK though

Reply:
ahhh, apparently I can't read. That makes sense, thanks.